### PR TITLE
[plugin_template] replaces deprecated method usage for android template

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description

Removes the usage of `getFlutterEngine` since it's deprecated, and uses `getBinaryMessenger` for java and `binaryMessenger` for kt to provide the instance of `BinaryMessenger` to `MethodChannel`. 
The PR also removes redundant semicolons and visibility modifier in kt files.

## Related Issues

#62130

## Tests

 I'm not exactly sure if this needs test. @blasten let me know if I should write test for these changes.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/templates/plugin/android-kotlin.tmpl/src/main/kotlin/androidIdentifier/pluginClass.kt.tmpl [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package {{androidIdentifier}}
 
-import androidx.annotation.NonNull;
+import androidx.annotation.NonNull
```
Comment (blasten) [inline packages/flutter_tools/templates/plugin/android-kotlin.tmpl/src/main/kotlin/androidIdentifier/pluginClass.kt.tmpl:3]: nice catch

**Comment by blasten** [comment]
The build side of things is tested in `dev/devicelab/bin/tasks/plugin_test.dart`. That should be sufficient in this case.

**Review by blasten — APPROVED** [review]
LGTM

**Comment by AyushBherwani1998** [comment]
> 
> 
> The build side of things is tested in `dev/devicelab/bin/tasks/plugin_test.dart`. That should be sufficient in this case.

Awesome!! Thanks

**Pull request merged**